### PR TITLE
fix(react-datepicker-compat): Make DatePicker Compat stories SSR safe

### LINKED ISSUE #27957: [Bug]: labels in stories are not SSR safe
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

None

### Reproduction

N/A

### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

None

### Reproduction

N/A

## Actual Behavior
SSR tests are failing

## Expected Behavior
SSR tests are passing

## Repro
- Open `scripts/test-ssr/src/commands/main.ts`
- Uncomment following
  ```diff
  const EXCLUDED_STORIES = [
  -  // TODO: re-enable ISSUE LINK
  - '**/react-datepicker-compat/**',
    // TODO: re-enable ISSUE LINK
    '**/react-toast/**',
    // Portals currently do not support hydration
    // https://github.com/facebook/react/issues/13097
    '**/react-portal/**',
    // https://github.com/microsoft/fluentui/issues/27338
    '**/react-table/stories/DataGrid/Virtualization.stories.tsx',
    '**/react-table/stories/Table/Virtualization.stories.tsx',
  ];
  ```
- Run tests `yarn workspace @fluentui/react-datepicker-compat test-ssr`
- Notice that tests are failing
- The problem is with a `Date` used in a story:
  https://github.com/microsoft/fluentui/blob/ab53a4c86a49a4d4268f3d87809bee263f818c5d/packages/react-components/react-datepicker-compat/stories/DatePicker/DatePickerErrorHandling.stories.tsx#L23-L26
  

### Logs

```shell
N/A
```

### Requested priority

High

### Products/sites affected

N/A

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### Logs

```shell
N/A
```

### Requested priority

High

### Products/sites affected

Fluent UI repo itself

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

Date boundaries story uses Date.now().

## New Behavior

Date boundaries story no longer uses Date.now().

## Related Issue(s)

- Fixes #27957

### CODE REVIEW

**Comment by sopranopillow** [comment]
This is the correct change, but this bug is coming up #27931. Once it's fixed this PR will be ready to merge.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                    |          Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr> | `221.766 kB`<br />`58.962 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d145902686da811aeb90e506b829044e1fb3cf90'>d145902686da811aeb90e506b829044e1fb3cf90</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Comment by sopranopillow** [comment]
@layershifter the error causing the crash (not the Date.now()) has been resolved and therefore the change is ready to be made!

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-datepicker-compat/stories/DatePicker/DatePickerDateBoundaries.stories.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ const useStyles = makeStyles({
   },
 });
 
-const today = new Date(Date.now());
+const today = new Date('05/24/2023');
```
Comment (layershifter) [inline packages/react-components/react-datepicker-compat/stories/DatePicker/DatePickerDateBoundaries.stories.tsx:11]: You need to modify also date formatting to make it deterministic i.e.:

(also in `DatePickerErrorHandling.stories.tsx`)

```ts
const today = new Date('05/24/2023');
const minDate = addMonths(today, -1);
const maxDate = addYears(today, 1);

function formatDate(date: Date) {
  return `${date.getMonth() + 1}/${date.getDate()}/${date.getFullYear()}`;
}
```

Why? Locales on a server and client might be different like in my case:

```
"The date boundaries for this example are 5/7/2023 to 6/7/2024." Client: "The date boundaries for this example are 07/05/2023 to 07/06/2024."
```

**Review by layershifter — APPROVED** [review]

**Review by Hotell — APPROVED** [review]

**Pull request merged**